web/src/app/main/components/NavSubMenu.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react'
+import { NavLink } from 'react-router-dom'
+import Collapse from '@material-ui/core/Collapse'
+import List from '@material-ui/core/List'
+import { makeStyles } from '@material-ui/styles'
+import ListItem from '@material-ui/core/ListItem'
+import ListItemText from '@material-ui/core/ListItemText'
+import ListItemIcon from '@material-ui/core/ListItemIcon'
+import Typography from '@material-ui/core/Typography'
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown'
+import { styles } from '../../styles/materialStyles'
+
+const useStyles = makeStyles(theme => {
+  const { nav } = styles(theme)
+  return {
+    nav,
+    subMenu: {
+      backgroundColor: theme.palette.primary['500'],
+      padding: '0',
+    },
+    link: {
+      textDecoration: 'none',
+      display: 'block',
+      color: '#fff',
+      '&:hover': {
+        textDecoration: 'none',
+      },
+    },
+    parentItem: {
+      color: theme.palette.primary['500'],
+    },
+    subMenuLinkText: {
+      paddingLeft: '3rem',
+    },
+    subMenuSelected: {
+      color: '#616161',
+      backgroundColor: '#ebebeb',
+      borderRight: '3px solid #D3D3D3',
+    },
+  }
+})
+
+export default function NavSubMenu(props) {
+  const { parentIcon, parentTitle, path, subMenuRoutes } = props
+  const [open, setOpen] = useState(false)
+  const classes = useStyles()
+
+  function handleClick(bool) {
+    setOpen(bool)
+  }
+
+  function renderParentLink(IconComponent, label) {
+    return (
+      <ListItem>
+        <ListItemIcon>
+          <IconComponent />
+        </ListItemIcon>
+        <ListItemText
+          className={classes.parentItem}
+          disableTypography
+          primary={<Typography variant='subtitle1'>{label}</Typography>}
+        />
+        <ArrowDropDownIcon className={classes.parentItem} />
+      </ListItem>
+    )
+  }
+
+  function renderSubMenu(subMenuRoutes) {
+    let subMenu = subMenuRoutes.map((route, key) => {
+      return (
+        <NavLink
+          exact
+          activeClassName={classes.subMenuSelected}
+          key={key}
+          className={classes.link}
+          to={route.path}
+        >
+          <ListItem>
+            <ListItemText className={classes.subMenuLinkText}>
+              {route.title}
+            </ListItemText>
+          </ListItem>
+        </NavLink>
+      )
+    })
+
+    return subMenu
+  }
+
+  function activeLink(match) {
+    if (!match) {
+      return handleClick(false)
+    }
+
+    return handleClick(true)
+  }
+  return (
+    <React.Fragment>
+      <NavLink to={path} className={classes.nav} isActive={activeLink}>
```

Comment:
This shouldn't result in a changing state, and does not represent a click. The default behavior is to match the path though, does it work if we simply omit it entirely?

Reply:
Are you talking about the isActive? We need it to change the state of the collapse component.

Reply:
The `Collapse in={}` bit shouldn't be using a state param, it should be calculated by the path directly.

Currently, this equates to the flow:
1. render
1. check if route matched
1. set state based on match
1. re-render with new state
1. re-check match
1. set-state (should be the same)
1. update collapsed

That causes a minimum of 2 renders and probably a flicker on slower systems. There's also no guarantee that `isActive` runs every time, or isn't cached -- as it is not a render hook, it's only meant to determine active state for the `NavLink`

Instead, just calculate the value based on the current route directly:
```js
<Collapse in={isThisOurRoute()}
```